wp-content/themes/pub/wporg-learn-2024/patterns/page-course-complete-content.php

```diff
@@ -5,6 +5,10 @@
  * Inserter: no
  */
 
+$course_id = isset( $_GET['course_id'] ) ? intval( $_GET['course_id'] ) : get_the_ID();
```

Comment:
Curious about this, did you find a case where `get_the_ID()` didn't work? I would have thought that would be the ideal means for getting it.

Reply:
I haven’t encountered this, but I remember being reminded before that in certain places like before/after the loop or in header, footer, or admin pages, `get_the_ID()` might not work, as it’s likely to return no value in those contexts. It works here because I saw [this snippet](https://github.com/WordPress/WordPress/blob/master/wp-includes/blocks/post-template.php#L101-L132) from the post_template block.